graphdatascience/procedure_surface/cypher/articulationpoints_cypher_endpoints.py

```diff
@@ -0,0 +1,308 @@
+        relationship_types: Optional[List[str]] = None,
+        node_labels: Optional[List[str]] = None,
+        sudo: Optional[bool] = None,
+        log_progress: Optional[bool] = None,
+        username: Optional[str] = None,
+        concurrency: Optional[Any] = None,
+        job_id: Optional[Any] = None,
+    ) -> ArticulationPointsMutateResult:
+        """
+        Executes the ArticulationPoints algorithm and writes the results to the in-memory graph as node properties.
+
+        ArticulationPoints is an algorithm that finds nodes that disconnect components if removed.
+        These nodes are critical for the connectivity of the graph.
+
+        Parameters
+        ----------
+        G : Graph
+            The graph to run the algorithm on
+        mutate_property : str
+            The property name to store the articulation point flag for each node
+        relationship_types : Optional[List[str]], default=None
+            The relationship types used to select relationships for this algorithm run
+        node_labels : Optional[List[str]], default=None
+            The node labels used to select nodes for this algorithm run
+        sudo : Optional[bool], default=None
+            Override memory estimation limits
+        log_progress : Optional[bool], default=None
+            Whether to log progress
+        username : Optional[str], default=None
+            The username to attribute the procedure run to
+        concurrency : Optional[Any], default=None
+            The number of concurrent threads
+        job_id : Optional[Any], default=None
+            An identifier for the job
+
+        Returns
+        -------
+        ArticulationPointsMutateResult
+            Algorithm metrics and statistics including the count of articulation points found
+        """
+        config = ConfigConverter.convert_to_gds_config(
+            mutate_property=mutate_property,
+            concurrency=concurrency,
+            job_id=job_id,
+            log_progress=log_progress,
+            node_labels=node_labels,
+            relationship_types=relationship_types,
+            sudo=sudo,
+            username=username,
+        )
+
+        # Run procedure and return results
+        params = CallParameters(graph_name=G.name(), config=config)
+        params.ensure_job_id_in_config()
+
+        cypher_result = self._query_runner.call_procedure(
+            endpoint="gds.articulationPoints.mutate", params=params
+        ).squeeze()
+
+        return ArticulationPointsMutateResult(**cypher_result.to_dict())
+
+    def stats(
+        self,
+        G: Graph,
+        relationship_types: Optional[List[str]] = None,
+        node_labels: Optional[List[str]] = None,
+        sudo: Optional[bool] = None,
+        log_progress: Optional[bool] = None,
+        username: Optional[str] = None,
+        concurrency: Optional[Any] = None,
+        job_id: Optional[Any] = None,
+    ) -> ArticulationPointsStatsResult:
+        """
```

Comment:
Thinking about it now, is it not enough that the documentation is on the base class?
Maybe we can and should remove the docs from all the implementations? WDYT?

Reply:
yes i like the idea to remove it from the impl